src/api/auth.ts

```diff
@@ -0,0 +1,8 @@
+export function generateRandomString(length: number) {
+	let result = '';
+	const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~';
+	for (let i = 0; i < length; i++) {
+		result += possible.charAt(Math.floor(Math.random() * possible.length));
+	}
+	return result;
```

Comment:
Is this Spotify's own unique way of wanting developers to generate a random string?

If not maybe try exploring something in like Crypto?
https://developer.mozilla.org/en-US/docs/Web/API/Crypto

Reply:
This method was taken from Spotify's own implementation example on their document:
https://github.com/spotify/web-api-examples/blob/master/authentication/authorization_code/app.js
I also noticed I had a small typo in the allowed characters which is used for the code verifier and not here.
Using crpyto will add a tiny bit of complexity but will achieve a cryptographically secure source of random values.

Reply:
If you just use something that works from Crypto, this entire PR isn't even necessary..